sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3173,6 +3179,9 @@ object Sequence {
          |  int $i = 0;
          |
          |  while ($t < $exclusiveItem ^ $stepSign < 0) {
+         |    if ($i == $arr.length) {
+         |      $arr = java.util.Arrays.copyOf($arr, $i + 1);
```

Comment:
Why the codegen and "interpret" code is different? If there is no specific reason, please, make them the same.

Reply:
I got it. This LGTM.

Reply:
Actually, you're right, they are different and I think the code genned code is more correct. I followed up with #37542